src/Module/FilterCartItemNameHandler.php

```diff
@@ -0,0 +1,136 @@
+use Dhii\Data\Container\NormalizeKeyCapableTrait;
+use Dhii\Exception\CreateInvalidArgumentExceptionCapableTrait;
+use Dhii\Exception\CreateOutOfRangeExceptionCapableTrait;
+use Dhii\Exception\CreateRuntimeExceptionCapableTrait;
+use Dhii\I18n\StringTranslatingTrait;
+use Dhii\Invocation\InvocableInterface;
+use Dhii\Iterator\CountIterableCapableTrait;
+use Dhii\Iterator\ResolveIteratorCapableTrait;
+use Dhii\Storage\Resource\SelectCapableInterface;
+use Dhii\Util\Normalization\NormalizeIntCapableTrait;
+use Dhii\Util\Normalization\NormalizeStringCapableTrait;
+use Psr\EventManager\EventInterface;
+
+/**
+ * The handler for filtering cart item names to just be the service names.
+ *
+ * @since [*next-version*]
+ */
+class FilterCartItemNameHandler implements InvocableInterface
+{
+    /* @since [*next-version*] */
+    use ContainerGetCapableTrait;
+
+    /* @since [*next-version*] */
+    use CountIterableCapableTrait;
+
+    /* @since [*next-version*] */
+    use ResolveIteratorCapableTrait;
+
+    /* @since [*next-version*] */
+    use NormalizeIntCapableTrait;
+
+    /* @since [*next-version*] */
+    use NormalizeKeyCapableTrait;
+
+    /* @since [*next-version*] */
+    use NormalizeStringCapableTrait;
+
+    /* @since [*next-version*] */
+    use CreateContainerExceptionCapableTrait;
+
+    /* @since [*next-version*] */
+    use CreateNotFoundExceptionCapableTrait;
+
+    /* @since [*next-version*] */
+    use CreateOutOfRangeExceptionCapableTrait;
+
+    /* @since [*next-version*] */
+    use CreateInvalidArgumentExceptionCapableTrait;
+
+    /* @since [*next-version*] */
+    use CreateRuntimeExceptionCapableTrait;
+
+    /* @since [*next-version*] */
+    use StringTranslatingTrait;
+
+    /**
+     * The services SELECT resource model.
+     *
+     * @since [*next-version*]
+     *
+     * @var SelectCapableInterface
+     */
+    protected $servicesSelectRm;
+
+    /**
+     * The expression builder.
+     *
+     * @since [*next-version*]
+     *
+     * @var object
+     */
+    protected $exprBuilder;
+
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param SelectCapableInterface $servicesSelectRm The services SELECT resource model.
+     * @param object                 $exprBuilder      The expression builder.
+     */
+    public function __construct(SelectCapableInterface $servicesSelectRm, $exprBuilder)
+    {
+        $this->servicesSelectRm = $servicesSelectRm;
+        $this->exprBuilder      = $exprBuilder;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function __invoke()
+    {
+        $event = func_get_arg(0);
+
+        if (!($event instanceof EventInterface)) {
+            throw $this->_createInvalidArgumentException(
+                $this->__('Argument is not an event instance'), null, null, $event
+            );
+        }
+
+        // Get service ID from event
+        $serviceId = $event->getParam(1);
```

Comment:
What happens if this parameter is not set?

Reply:
Service ID is null, query returns no services and the method returns without error. Basically it would be the equivalent of the Download ID not corresponding to a Download that has bookings enabled.

**Also note**: if the param is ever in fact _not set_, it would be due to a BC breaking change from EDD's end. Which might be possible after EDD 3.0.

Reply:
So, wouldn't it be better to check if it is actually set and not null, and avoid the whole query altogether, which would also save us from the possible BC break?

Reply:
Done.